Widgets/Bars/ClassBar.lua

```diff
@@ -250,6 +255,7 @@ end
 
 ---@param self ClassBarWidget
 local function ShouldShow(self)
+    if self.hideOutOfCombat and not UnitAffectingCombat("player") then return false end
```

Comment:
```suggestion
    if self.hideOutOfCombat and (not UnitAffectingCombat("player") or event == "PLAYER_REGEN_ENABLED") then return false end
```
This function should most likely be parsed `event` and compare against `PLAYER_REGEN_ENABLED`, since you can't always rely on `UnitAffectingCombat` providing the proper return.

Reply:
I have adjusted it but the correct event was `PLAYER_REGEN_DISABLED` (entered combat)